### ISSUE TITLE
App does not receive text messages and then crashes.

### ISSUE THREAD

**Issue by myfirstnameispaul** [juha-h/baresip-studio#244]
I am no longer able to receive text messages with this app.

I updated to the latest version on F-droid and the app now cannot receive text messages, and best I can tell it crashes when a text message is received.

**Comment by myfirstnameispaul** [comment]
It is baresip version 44.0.0 on Android 11/Lineage OS 18.1.

**Comment by juha-h** [comment]
I just installed baresip 44.0.0 from F-Droid to my Android 10 device and both sending and receiving of text message worked fine.  Can you show what `adb logcat | grep Baresip` gives when baresip receives the message.

**Comment by myfirstnameispaul** [comment]
Here is the crash:

```text
04-04 12:57:51.862 13839 13884 D Baresip Lib: got message 'I can crash you.' from peer 'sip:<redacted>@sanjose2.voip.ms'
04-04 12:57:51.866 13839 13885 D Baresip Lib: <redacted>@sanjose2.voip.ms: selected for <redacted>
04-04 12:57:52.265 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService.baresipStart(Native method)
04-04 12:57:52.265 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService.onStartCommand$lambda-2(BaresipService.kt:398)
04-04 12:57:52.265 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService.$r8$lambda$pKHUIHiUAqfLgwNror59bB61wDk(BaresipService.kt:-1)
04-04 12:57:52.265 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService$$ExternalSyntheticLambda4.run(D8$$SyntheticClass:-1)
04-04 12:57:52.268 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService.baresipStart(Native method)
04-04 12:57:52.268 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService.onStartCommand$lambda-2(BaresipService.kt:398)
04-04 12:57:52.268 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService.$r8$lambda$pKHUIHiUAqfLgwNror59bB61wDk(BaresipService.kt:-1)
04-04 12:57:52.268 13839 13884 F .tutpro.baresi: runtime.cc:655]   at com.tutpro.baresip.BaresipService$$ExternalSyntheticLambda4.run(D8$$SyntheticClass:-1)
04-04 12:57:53.056   737  1662 W ActivityManager: Scheduling restart of crashed service com.tutpro.baresip/.BaresipService in 1000ms for start-requested
```

Let me know if you want to see the restart.

**Comment by juha-h** [comment]
The above logcat output is from after the crash when Android tries to restart baresip.  Keep watching with `adb logcat | grep Baresip` and then send the message to baresip.  That shows in log what happened.  Also, before that turn on Debug in baresip Settings before you do that.

**Comment by juha-h** [comment]
Looks like I was able to reproduce the crash on my 32 bit ARMv7 device.  Does your device have 32 bit ARMv7 architecture?

**Comment by juha-h** [comment]
I found and fixed the bug.  It was ARMv7 related. Fix will be in version 44.1.1 or greater.  Thanks for reporting.

**Comment by juha-h** [comment]
The fix in in version 44.2.0.

**Comment by myfirstnameispaul** [comment]
Will this be arriving on F-droid?

**Comment by juha-h** [comment]
Paul writes:

> Will this be arriving on F-droid?

Yes, but unfortunately there usually is 4-5 day delay.  If you want, I
can create an apk and make it available for download.

**Comment by myfirstnameispaul** [comment]
I was thinking I would downgrade, but those versions don't install and I'm not clear if uninstalling removes the account data. I have something coming where I do need the text messages working on the phone (I also can use my provider website, but it is unusable on mobile).

**Comment by juha-h** [comment]
Unfortunately F-Droid build of 44.2.0 failed on F-Droid and I don't know why.  Since it may take many days to get it working again (if ever), I placed baresip 44.2.0 APK available here: https://box.tutpro.com/tmp/baresip-44.2.0.apk.  Backup baresip before install and then restore after install.

**Comment by myfirstnameispaul** [comment]
Is there a way to disable SMS in baresip? I just discovered my provider has an SMS app in F-droid that is working very well (if it was there before, then it had some kind of problem, but I don't remember now.)

**Comment by juha-h** [comment]
No, bareslib when it sends REGISTER request, always includes MESSAGE in Allow header.

**Comment by gdt** [comment]
But surely the messages in SIP are not SMS??

**Comment by juha-h** [comment]
Greg Troxel writes:

> But surely the messages in SIP are not SMS??

No, unless there is SIP MESSAGE/SMS gateway somewhere in between.

-- Juha

**Comment by juha-h** [comment]
Latest published versions of breasip (44.2.0) and baresip+ (31.2.0) should fix the crash.  Re-open if still an issue.
